### ISSUE TITLE
[ARCHETYPE-562] Error ordering Default Properties

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-archetype#829]
**[David Castro](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=dcasagu)** opened **[ARCHETYPE-562](https://issues.apache.org/jira/browse/ARCHETYPE-562?redirect=false)** and commented

Default properties are being ordered consediring the "exact" use of another property. The assumption implies that a property that uses a derived string of a previous property may not be ordered as expected:

 

```java
<requiredProperty key="domainUpperCase">
     <defaultValue>${${domainName}.toUpperCase()}</defaultValue>
     <validationRegex>[A-Z]+</validationRegex>
</requiredProperty>
```

 

 

 

---

**Affects:** 3.0.1, 3.1.0

**Remote Links:**
- [GitHub Pull Request #26
](https://github.com/apache/maven-archetype/pull/26)

**Comment by jira-importer** [comment]
**[David Castro](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=dcasagu)** commented

https://github.com/apache/maven-archetype/pull/25

**Comment by jira-importer** [comment]
**[Matthew Jason Benson](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=mbenson)** commented

[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o) I believe this was addressed by my commits:

* https://github.com/apache/maven-archetype/commit/137461e8dd71a26f53d25aa9c58e29e04127d6a7
* https://github.com/apache/maven-archetype/commit/54b5fb4bf16d70fce0ab3493d2e178b317a2f5b7

which enhanced `DefaultArchetypeGenerationConfigurator.RequiredPropertyComparator` to consider required properties referenced by other required properties' (complex) default expressions. I believe this issue should have been included in the 3.2.1 changelog as ERH intended.

**Comment by jira-importer** [comment]
**[Elliotte Rusty Harold](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=elharo)** commented

Please reopen if this is still an issue, but last comment suggests it's fixed
